lib/perl/Genome/VariantReporting/Suite/BamReadcount/VafInterpreterHelpers.pm

```diff
@@ -74,16 +74,79 @@ sub many_samples_field_descriptions {
 }
 sub per_sample_vaf_headers {
     my $component = shift;
-    return $component->create_sample_specific_field_names(
-        [vaf_fields()],
+    return sort_vaf_headers(
+        headers      => [$component->create_sample_specific_field_names([vaf_fields()])],
+        sample_names => [$component->sample_names],
     );
 }
 
 sub per_library_vaf_headers {
     my $component = shift;
-    return $component->create_library_specific_field_names(
-        [vaf_fields()],
+    return sort_vaf_headers(
+        headers       => [$component->create_library_specific_field_names([vaf_fields()])],
+        sample_names  => [$component->sample_names],
+        library_names => [$component->library_names],
     );
 }
 
+sub sort_vaf_headers {
+    my %params = @_;
+    return map { $_->{header} } sort {
+        $a->{label_priority} cmp $b->{label_priority} ||
+        $a->{sample} cmp $b->{sample} ||
+        $a->{library} cmp $b->{library} ||
+        $a->{vaf_priority} cmp $b->{vaf_priority}
+    } annotate_headers(%params);
+}
+
+my %VAF_CONVERSIONS = (
+    '_ref_count$' => 1,
+    '_var_count$' => 2,
+    '_vaf$'       => 3,
+);
+
+my %LABEL_CONVERSIONS = (
+    '^Normal'    => 1,
+    '^Discovery' => 2,
+    '^Followup'  => 3,
+);
+
+sub annotate_headers {
+    my %params        = @_;
+    my $headers       = $params{headers};
+    my $sample_names  = $params{sample_names};
+    my $library_names = $params{library_names};
+
+    my @annotated_headers;
+    for my $header (@$headers) {
+        my %annotated_header = (
+            header => $header,
+            library => 0,
```

Comment:
Fair enough! Maybe this subroutine should die if they're not present as expected.